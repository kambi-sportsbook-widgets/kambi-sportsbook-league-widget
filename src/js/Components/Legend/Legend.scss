:local(.legend) {
   display: flex;
   padding: 0 16px;
   margin: 10px 0;
   flex-wrap: wrap;

   li {
      flex: 1;
      list-style: none;
      display: flex;
      align-items: center;
      margin: 10px 20px 10px 0;

      :local(.circle) {
         flex: 0;
         line-height: 0;
         margin-right: 8px;
      }

      p {
         flex: 1;
         color: #999;
         font-size: 12px;
         margin: 0;
         line-height: 14px;
         min-width: 100px;
      }
   }
}
