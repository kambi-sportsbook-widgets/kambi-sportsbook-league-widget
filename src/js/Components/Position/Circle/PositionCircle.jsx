import React from 'react';
import PropTypes from 'prop-types';
import styles from './PositionCircle.scss';

const COLORS = {
   GREEN: {
      background: 'KambiWidget-winPrimaryColor dark',
      text: 'KambiWidget-winSecondaryColor',
   },
   LIGHT_GREEN: {
      background: 'KambiWidget-winPrimaryColor light',
      text: 'KambiWidget-winSecondaryColor',
   },
   ORANGE: {
      background: 'KambiWidget-drawPrimaryColor dark',
      text: 'KambiWidget-drawSecondaryColor',
   },
   LIGHT_ORANGE: {
      background: 'KambiWidget-drawPrimaryColor light',
      text: 'KambiWidget-drawSecondaryColor',
   },
   RED: {
      background: 'KambiWidget-losePrimaryColor dark',
      text: 'KambiWidget-loseSecondaryColor',
   },
   LIGHT_RED: {
      background: 'KambiWidget-losePrimaryColor light',
      text: 'KambiWidget-loseSecondaryColor',
   },
   TRANSPARENT: {
      background: 'transparent',
      text: 'KambiWidget-card-text-color',
   },
};

const PositionCircle = ({ color, children }) => {
   let backgroundCss = `${styles.wrapper}`;
   let textCss = '';
   if (color == null) {
      color = 'TRANSPARENT';
   }
   backgroundCss += ` ${COLORS[color].background}`;
   textCss += ` ${COLORS[color].text}`;

   return (
      <span className={backgroundCss}>
         <span className={textCss}>
            {children}
         </span>
      </span>
   );
}

PositionCircle.propTypes = {

   /**
    * Current participant position
    */
   color: PropTypes.oneOf(Object.keys(COLORS)),

   /**
    * Optional node te be displayed inside circle
    */
   children: PropTypes.node

};

PositionCircle.defaultProps = {
   color: null
};

PositionCircle.COLORS = COLORS;

export default PositionCircle;
