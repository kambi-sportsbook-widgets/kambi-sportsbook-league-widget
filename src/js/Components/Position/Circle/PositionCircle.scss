:local(.wrapper) {
   border-radius: 50%;
   width: 22px;
   height: 22px;
   display: inline-block;
   background: #fff;
   line-height: 0;

   &.no-circle {
      background: transparent;
   }
}

:local(.circle) {
   border-radius: 50%;
   width: 22px;
   height: 22px;
   display: inline-block;
   line-height: 21px;
   text-align: center;
   font-size: 12px;

   &.light {
      opacity: .6;
   }

   &.transparent {
      background: transparent;
   }
}
