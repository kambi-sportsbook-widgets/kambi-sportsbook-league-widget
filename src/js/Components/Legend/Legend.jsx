import React from 'react'
import PropTypes from 'prop-types'
import { translationModule } from 'kambi-widget-core-library'
import styles from './Legend.scss'
import PositionCircle from '../Position/Circle/PositionCircle'

const t = translationModule.getTranslation.bind(translationModule)
/**
 * Removes duplicate legends based on matching descriptions
 */
const Legend = ({ items }) => {
  items = items.filter(
    (item, i, arr) =>
      i === arr.findIndex(t => t.description === item.description)
  )

  return (
    <ul className={styles.legend}>
      {items.map((item, i) => (
        <li key={i}>
          <div className={styles.circle}>
            <PositionCircle color={item.color} />
          </div>
          <p>{t(item.description)}</p>
        </li>
      ))}
    </ul>
  )
}

Legend.propTypes = {
  /**
   * Array of legend items
   */
  items: PropTypes.arrayOf(
    PropTypes.shape({
      color: PropTypes.string.isRequired,
      description: PropTypes.string.isRequired,
    })
  ).isRequired,
}

export default Legend
