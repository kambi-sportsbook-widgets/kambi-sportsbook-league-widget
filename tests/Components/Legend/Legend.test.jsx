/* eslint-env jest */
import React, { Children } from 'react';
import Legend from '../../../src/js/Components/Legend/Legend';
import PositionCircle from '../../../src/js/Components/Position/Circle/PositionCircle';
import ReactTestRenderer from 'react-test-renderer';

describe('Legend DOM rendering', () => {

   it('renders correctly with empty items list', () => {
      const tree = ReactTestRenderer.create(
         <Legend items={[]} />
      ).toJSON();

      expect(tree).toMatchSnapshot();
   });

   it('renders correctly with items', () => {
      const items = Object.keys(PositionCircle.COLORS).map((color, i) => ({color, description: `${i}th item`}));

      const tree = ReactTestRenderer.create(
         <Legend items={items} />
      ).toJSON();

      expect(tree).toMatchSnapshot();
   });

});
