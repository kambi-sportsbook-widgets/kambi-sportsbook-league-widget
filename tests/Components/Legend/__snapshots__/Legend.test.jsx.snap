// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Legend DOM rendering renders correctly with empty items list 1`] = `
<ul
  className="legend"
>
  
</ul>
`;

exports[`Legend DOM rendering renders correctly with items 1`] = `
<ul
  className="legend"
>
  <li>
    <div
      className="circle"
    >
      <PositionCircle
        color="GREEN"
      />
    </div>
    <p>
      0th item
    </p>
  </li>
  <li>
    <div
      className="circle"
    >
      <PositionCircle
        color="LIGHT_GREEN"
      />
    </div>
    <p>
      1th item
    </p>
  </li>
  <li>
    <div
      className="circle"
    >
      <PositionCircle
        color="ORANGE"
      />
    </div>
    <p>
      2th item
    </p>
  </li>
  <li>
    <div
      className="circle"
    >
      <PositionCircle
        color="LIGHT_ORANGE"
      />
    </div>
    <p>
      3th item
    </p>
  </li>
  <li>
    <div
      className="circle"
    >
      <PositionCircle
        color="RED"
      />
    </div>
    <p>
      4th item
    </p>
  </li>
  <li>
    <div
      className="circle"
    >
      <PositionCircle
        color="LIGHT_RED"
      />
    </div>
    <p>
      5th item
    </p>
  </li>
  <li>
    <div
      className="circle"
    >
      <PositionCircle
        color="TRANSPARENT"
      />
    </div>
    <p>
      6th item
    </p>
  </li>
</ul>
`;
